doc/readme.md

```diff
@@ -0,0 +1,362 @@
+
+*Hinweis:* Die Game-Loop wird automatisch ausgeführt, Sie müssen sie nicht aktiv aufrufen.
+
+`Game` implementiert noch weitere wichtige Methoden:
+
+- `setup` wird zu Beginn der Anwendung aufgerufen. In dieser Methode werden die Objekte (wie die Systeme) initialisiert und konfiguriert, welche bereits vor dem Spielstart existieren müssen. In der Vorgabe wird hier bereits das erste Level geladen, die Systeme angelegt und der Held initialisiert.
+- `frame` wird in jedem Frame einmal aufgerufen.
+- `onLevelLoad` wird immer dann aufgerufen, wenn ein Level geladen wird. Hier werden später Entitäten erstellt, die initial im Level verteilt werden.
+- `getEntities` liefert das HashSet `entities` zurück. `entities` speichert alle Entitäten, welche sich momentan im Dungeon befinden.
+- `addEntity` fügt neue Entitäten im nächsten Frame in das Spiel ein.
+- `removeEntity` entfernt Entitäten im nächsten Frame aus dem Spiel.
+- `main` startet das Spiel.
+
+
+### Entity
+
+Entitäten sind die Objekte im Spiel. Im Code sind sie nur leere Container, dessen Eigenschaften über die zugewiesenen Components bestimmt werden. Entitäten haben neben den Components keine eigenen Attribute oder Funktionen.
+
+Sie werden nicht durch Ableiten der Klasse `Entity` erzeugt (Ausnahme ist der Held, dazu später mehr). Stattdessen werden sie durch Erzeugen einer Instanz von `Entity` und Hinzufügen von`Component`s implementiert.
+
+
+### Component
+
+Die `Component`s sind die Datensätze der Entitäten und beschreiben dessen Eigenschaften. Eine Component-Instanz gehört zu genau einer Entität-Instanz.
+
+Jedes Component muss von der abstrakten Klasse `Component` abgeleitet werden.
+Schauen Sie sich auch die [bereits implementierten Components](ecs/components/readme.md) an.
+
+
+### ECS_System
+
+Systeme agieren auf die Components und ändern die Werte in diesen. Sie bestimmen also das Verhalten der Entitäten.
+Ein System kann auf mehrere Components agieren.
+
+Um ein eigenes System zu erstellen, muss von der abstrakten Klasse `ECS_System` abgeleitet werden.
+Schauen Sie sich auch die [bereits implementierten Systeme](ecs/systems/readme.md) an.
+
+*Hinweis:* Um mehr darüber zu erfahren, wie Sie eigene Entitäten, Components und Systeme erstellen, werfen Sie einen Blick in [Eigene Inhalte erstellen](ecs/create_own_content.md).
+
+
+## Übung: Eigenen Helden erstellen
+
+Zwar gibt es in den Vorlagen bereits einen Helden (den schauen wir uns am Ende dieses Kapitels genauer an), trotzdem wird Ihnen hier erklärt, wie Sie Ihre erste eigene Entität in das Spiel implementieren.
+
+*Hinweis:* Wenn Sie mitprogrammieren wollen, löschen Sie in `starter.Game` die Zeile 118 (`hero = new Hero(new Point(0, 0));`) und das Import-Statement `import ecs.entities.Hero;`.
```

Comment:
Sollte man hier auf die Zeilenangabe vielleicht besser verzichten? Die könnte bei einer kleinen Änderung schnell nicht mehr aktuell sein.

Reply:
Hatte ich auch schon drüber nachgedacht.

Vorteil: Die Studis finden das schneller/leichter.
Nachteil: Jede kleine Änderung wirft das über den Haufen. 

**Vielleicht ist ein besseres Wording**: "..., kommentieren Sie in der Methode `starter.Game#setup()` die Zeile ``hero = new Hero();`` aus.".

Reply:
- Auf das Import-Statement würde ich hier nicht eingehen, das spielt ja keine Rolle. 
- Und _löschen_ würde ich die Zeile nicht, weil ich da ja dann meinen eigenen Helden einbaue. 
- Und das mit dem `new Point()` hat sich ja mittlerweile auch erledigt - das ist im Default-Ctor eingebaut, kann hier also weg.